File: Dante1/Module1.04/Zadanie4.26/main.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int error;

    int array_main[1000];
    int size_of_array_main = 0;
    printf("Enter numbers: ");

    for(int i = 0; i < 1000; i++)
    {
        error = scanf("%i", &array_main[i]);
        if(error != 1)
        {
            printf("Incorrect input");
            return 1;
        }
        if(array_main[i] == 0)
        {
            break;
        }
        size_of_array_main++;
    }

    int array_support[1000];
    int size_of_array_support = 0;
    printf("\nEnter numbers: ");

    for(int i = 0; i < 1000; i++)
    {
        error = scanf("%i",&array_support[i]);
        if(error != 1)
        {
            printf("Incorrect input");
            return 1;
        }
        if(array_support[i] == 0)
        {
            break;
        }
        size_of_array_support++;
    }

    if(size_of_array_main > 100)
    {
        size_of_array_main = 100;
    }
    if(size_of_array_main < 2)
    {
        printf("Not enough data available");
        return 2;
    }
    if(size_of_array_support == 0)
    {
        printf("Not enough data available");
        return 2;
    }
    if(size_of_array_support != 2)
    {
        printf("Incorrect input");
        return 1;
    }

    int counter = 0;
    for(int i = 0; i < size_of_array_main - 1; i++)
    {
        if(array_main[i] == array_support[0])
        {
            if(array_main[i + 1] == array_support[1])
            {
                counter++;
            }
        }
    }

    printf("\n%i",counter);
    for(int i = 0; i < size_of_array_main - 1; i++)
    {
        if(array_main[i] == array_support[0])
        {
            if(array_main[i + 1] == array_support[1])
            {
                printf("\n%i", i);
            }
        }
    }

    return 0;
}

File: Dante1/Module1.04/Zadanie4.26/main.zig
fn main() {
    let mut input: String = String::new();

    let mut vector: [i32; 1000] = [0; 1000];
    let mut size: usize = 0;

    println!("Enter: ");

    for i in 0..2000 {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Error");
        vector[size] = input.trim().parse().expect("Error");

        if vector[size] == 0 {
            break;
        }

        size += 1;
    }

    let mut vector2: [i32; 1000] = [0; 1000];
    let mut size2: usize = 0;

    println!("Enter: ");

    for i in 0..2000 {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Error");
        vector2[size2] = input.trim().parse().expect("Error");

        if vector2[size2] == 0 {
            break;
        }

        size2 += 1;
    }

    if size > 100 {
        size = 100;
    }
    if size < 2 {
        panic!("Not enough data available");
    }
    if size2 == 0 || size != 2{
        panic!("Not enough data available");
    }

    let mut counter: i32 = 0;
    for i in 0..size - 1 {
        if vector[i] == vector2[0] {
            if vector[i + 1] == vector2[1] {
                counter += 1;
            }
        }
    }

    println!("{}", counter);

    for i in 0..size - 1 {
        if vector[i] == vector2[0] {
            if vector[i + 1] == vector2[1] {
                println!("\n{}", i);
            }
        }
    }
}
